packages/vhd-lib/parseVhdStream.js

```diff
@@ -13,124 +14,246 @@ const cappedBufferConcat = (buffers, maxSize) => {
   return buffer
 }
 
-exports.parseVhdStream = async function* parseVhdStream(stream) {
-  let bytesRead = 0
+function readLastSector(stream) {
+  return new Promise((resolve, reject) => {
+    let bufFooterEnd = Buffer.alloc(0)
+    stream.on('data', chunk => {
+      if (chunk.length > 0) {
+        bufFooterEnd = cappedBufferConcat([bufFooterEnd, chunk], SECTOR_SIZE)
+      }
+    })
 
-  // handle empty space between elements
-  // ensure we read stream in order
-  async function read(offset, size) {
-    assert(bytesRead <= offset, `offset is ${offset} but we already read ${bytesRead} bytes`)
-    if (bytesRead < offset) {
+    stream.on('end', () => resolve(bufFooterEnd))
+    stream.on('error', reject)
+  })
+}
+
+class StreamParser {
+  #bufFooter
```

Comment:
Isn't it problematic like `#readBlockData`? :thinking:

Reply:
it only fails on private method but works on private properties 
Do you want me to use `_` everywhere for coherence ?

Reply:
Weird.

I don't have any opinion on this, do as you wish :slightly_smiling_face:

Reply:
let's keep it as is